File: BaseLua/src/main/java/org/anime_game_servers/lua/models/BooleanLuaValue.java
package org.anime_game_servers.lua.models;

public class BooleanLuaValue extends MockLuaValue {
    public static final BooleanLuaValue TRUE = new BooleanLuaValue(true);
    public static final BooleanLuaValue FALSE = new BooleanLuaValue(false);
    private final boolean value;
    public BooleanLuaValue(boolean value){
        this.value = value;
    }

    @Override
    public boolean isBoolean() {
        return true;
    }

    @Override
    public boolean asBoolean() {
        return value;
    }

    @Override
    public int asInteger() {
        return value ? 1 : 0;
    }

    @Override
    public long asLong() {
        return value ? 1 : 0;
    }

    @Override
    public double asDouble() {
        return value ? 1 : 0;
    }

    @Override
    public float asFloat() {
        return value ? 1 : 0;
    }

    @Override
    public String asString() {
        return value ? "true" : "false";
    }

    @Override
    public <T> T asObject(Class<T> type) {
        if(type == Boolean.class)
            return type.cast(value);
        return null;
    }
}

File: base/src/commonMain/kotlin/org/anime_game_servers/lua/models/BooleanLuaValue.kt
package org.anime_game_servers.lua.models

import org.anime_game_servers.lua.engine.Class
import kotlin.jvm.JvmField

open class BooleanLuaValue(private val value: Boolean) : MockLuaValue {
    override fun isBoolean() = true

    override fun asBoolean() = value
    override fun asInteger() = if (value) 1 else 0
    override fun asLong() = if (value) 1L else 0L
    override fun asDouble() = if (value) 1.0 else 0.0
    override fun asFloat() = if (value) 1f else 0f
    override fun asString() = value.toString()

    override fun <T> asObject(type: Class<T>): T? = null

    companion object {
        @JvmField
        val TRUE = BooleanLuaValue(true)
        @JvmField
        val FALSE = BooleanLuaValue(false)
    }
}
